### DISCUSSION TITLE
How to disable refinement algorithms？

### DISCUSSION THREAD

**Discussion by sonw7** [gwlucastrig/Tinfour#101] [Q&A]
In practical applications, the addition of composite points for refinement affects the original data. Therefore, I would like to ask where you can set up and disable the refinement function, using only the most primitive slicing function.

**Comment by gwlucastrig** [comment]
Well, you're in luck.  I haven't yet implemented the Delaunay Refinement algorithms.  So refinement is never applied.  When I do finish the work, the default operation will be for the Delaunay Triangulation (TIN) to be run without refinement unless the calling application explicitly enables it.

**Comment by gwlucastrig** [comment]
Looking at your previous question, (Discussion #100), I think I might not have understood what you are driving at and I was unclear in my use of terminology.

Currently, synthetic points are added only if the application adds constraints to the triangulation, and only if the application indicates that it wants to restore the Delaunay compliance to the result.  You control whether that happens by setting the restoreConformity option to true.

The term "Delaunay Refinement", at least as I use it,  refers to a different technique.  In Delaunay Refinement, you have a Delaunay Triangulation with skinny triangles (such as those that occur near boundaries) and you improve the shape of the triangles by inserting artificial "Steiner points" at optimal locations.  I have wanted to implement this feature for a long time, but I have not been able to do so.  I am hoping to have it working early in 2024.
